app/invocation/child_invocation_card.tsx

```diff
@@ -1,60 +1,57 @@
 import React from "react";
 import format from "../format/format";
-import router from "../router/router";
 import { build_event_stream } from "../../proto/build_event_stream_ts_proto";
+import { CheckCircle, PlayCircle, XCircle, CircleSlash, Timer } from "lucide-react";
+import Link from "../components/link/link";
+
+export type CommandStatus = "failed" | "succeeded" | "in-progress" | "queued" | "not-run";
 
 export type BazelCommandResult = {
-  invocation:
-    | build_event_stream.WorkflowConfigured.IInvocationMetadata
-    | build_event_stream.ChildInvocationsConfigured.IInvocationMetadata;
+  status: CommandStatus;
+  invocation: InvocationMetadata;
   durationMillis?: number;
 };
 
+export type InvocationMetadata =
+  | build_event_stream.WorkflowConfigured.IInvocationMetadata
+  | build_event_stream.ChildInvocationsConfigured.IInvocationMetadata;
+
 export type ChildInvocationCardProps = {
-  status: string;
-  results: BazelCommandResult[];
-  className: string;
-  icon: JSX.Element;
-  linksDisabled?: boolean;
+  result: BazelCommandResult;
 };
 
 export default class ChildInvocationCard extends React.Component<ChildInvocationCardProps> {
-  private handleCommandClicked(invocationId: string, e: React.MouseEvent) {
-    // TODO(siggisim): Switch this to using the <Link> component
-    if (e.metaKey || e.ctrlKey) {
-      return;
+  private isClickable() {
+    return this.props.result.status !== "queued" && this.props.result.status !== "not-run";
+  }
+
+  private renderStatusIcon() {
+    switch (this.props.result.status) {
+      case "succeeded":
+        return <CheckCircle className="icon" />;
+      case "failed":
+        return <XCircle className="icon" />;
+      case "in-progress":
+        return <PlayCircle className="icon" />;
+      case "queued":
+        return <Timer className="icon" />;
+      case "not-run":
+        return <CircleSlash className="icon" />;
```

Comment:
Consider adding a default case that shows some kind of error icon

Reply:
by default it'll fall through and return undefined, rendering nothing, which seems OK to me since it shouldn't happen in practice (we compute these statuses in the UI rather than the backend) - but I made it a bit less magical by adding an explicit `default:` case that returns undefined.